### PR TITLE
WT-8905 Refactor test_tiered* python tests

### PR DESCRIPTION
**Pull request by ruby-oujo**
Refactored the test_tiered* python tests

- Common code was moved into `helper_tiered.py` and common functions were set up 
- Common functions include `get_conn_config`, `get_check`
- Creating scenarios using the `tiered_storage_sources` list was moved and made global in `helper_tiered.py`
- Utilised class `TieredConfigMixin` from [WT-9026](https://jira.mongodb.org/browse/WT-9026) to refactor out `conn_extensions` and inherit this from the parent class. 
- Created `get_bucket_name` function to take index of buckets in a list. 
- Checked for unused imports
- Merged changes from:
  - [WT-9029](https://jira.mongodb.org/browse/WT-9029) (Remove object_target_size option) 
  - [WT-9122](https://jira.mongodb.org/browse/WT-9122) Remove support for internal tiered manager thread
  - With these changes there is less variability in the `conn_config` and may need to revisit the usage of a helper function `get_conn_config` as opposed to inheriting from `TieredConfigMixin`.

### CODE REVIEW

**Review by ddanderson — APPROVED** [review]
This is nice work.  One small suggestion for improved readability.

File: test/suite/test_tiered02.py [hunk 1]
```diff
@@ -54,33 +43,16 @@ class test_tiered02(wttest.WiredTigerTestCase):
     ]
 
     # Make scenarios for different cloud service providers
-    scenarios = make_scenarios(storage_sources, complex_dataset)
+    scenarios = make_scenarios(tiered_storage_sources[:2], complex_dataset)
```
[the remark this answers is not in the document]
Comment (ruby-oujo) [reply test/suite/test_tiered02.py]: Hi Don, I created a slice in `helper_tiered.py` and saved it in a variable called `storage_sources` and left a comment above the new variable to explain the use of the sublist. In the other test files I have changed this to all use the variable  `storage_sources`, hope this improves the readability.  I have also used `tiered_storage_dirstore_source` to save the sublist in the case where only that storage source is required.

File: test/suite/helper_tiered.py [hunk 2]
```diff
@@ -55,30 +55,42 @@ def get_auth_token(storage_source):
...
+        storage_source.assertEqual(tc[str(i)], str(i))
+    tc.set_key(str(n))
+    storage_source.assertEquals(tc.search(), wiredtiger.WT_NOTFOUND)
 
 # Generate a unique object prefix for the S3 store. 
 def generate_s3_prefix(test_name = ''):
     # Generates a unique prefix to be used with the object keys, eg:
-    # "s3test/2022-31-01-16-34-10_623843294/"
-    prefix = 's3test/'
+    # "s3test_artefacts--python_2022-31-01-16-34-10_623843294/"
```
Comment (tammybailey) [inline test/suite/helper_tiered.py:89]: Please change this back to `s3test/` (or `s3test/python` if there is a need to distinguish from unit tests). We don't need to use something this complicated and strange.

Also, please add a comment that the objects prefixed with `s3test/*` are deleted after X days according to an AWS retention policy on the S3 bucket, and note who to speak with (e.g. "release manager") if changes to the object prefix or retention policy are required.

File: test/suite/test_tiered02.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ddanderson) [reply test/suite/test_tiered02.py]: Yes that helps, thanks!

File: test/suite/helper_tiered.py (hunk 2 above)
Comment (ruby-oujo) [reply test/suite/helper_tiered.py]: Hi Tammy 🙂 , I have changed the prefix pattern here to be back to `s3test/python/`  and I have also changed the prefix pattern in the unit testing to `s3test/` to match this pattern for the lifecycle rule / retention policy. I have added a comment in both places the prefixes are used and have put "release manager" for who to speak with in the interim until we confirm or decide on anyone else more suitable to contact. Hope this helps!

File: ext/storage_sources/s3_store/test/test_s3_connection.cpp [hunk 3]
```diff
@@ -35,7 +35,11 @@ const Aws::String region = Aws::Region::AP_SOUTHEAST_2;
 const double throughputTargetGbps = 5;
 const uint64_t partSize = 8 * 1024 * 1024;  // 8 MB.
 static std::string bucketName("s3testext"); // Can be overridden with environment variables.
-static std::string objPrefix("s3test_artefacts--unit_"); // To be concatenated with a random string.
+
+// Objects with the prefex pattern "s3test/*" are deleted after a certain period of time according
+// to the lifecycle rule on the S3 bucket. Should you wish to make any changes to the prefix pattern
+// or lifecycle of the object, please speak to the release manager.
+static std::string objPrefix("s3test/"); // To be concatenated with a random string.
```
Comment (tammybailey) [inline ext/storage_sources/s3_store/test/test_s3_connection.cpp:42]: Should this be `s3test/unit/`?

File: test/suite/helper_tiered.py (hunk 2 above)
Comment (tammybailey) [reply test/suite/helper_tiered.py]: Looks good! Thank you :)

**Review by tammybailey — APPROVED** [review]
One small question/comment, else LGTM. Nice work!

File: ext/storage_sources/s3_store/test/test_s3_connection.cpp (hunk 3 above)
Comment (ruby-oujo) [reply ext/storage_sources/s3_store/test/test_s3_connection.cpp:42]: I considered this, it does make sense to match the pattern more closely to the python tests. I will add this in now!

File: test/suite/helper_tiered.py [hunk 4]
```diff
@@ -55,27 +55,42 @@ def get_auth_token(storage_source):
...
+        storage_source.assertEqual(tc[str(i)], str(i))
+    tc.set_key(str(n))
+    storage_source.assertEquals(tc.search(), wiredtiger.WT_NOTFOUND)
 
 # Generate a unique object prefix for the S3 store. 
 def generate_s3_prefix(test_name = ''):
     # Generates a unique prefix to be used with the object keys, eg:
-    # "s3test/2022-31-01-16-34-10_623843294/"
-    prefix = 's3test/'
+    # "s3test_artefacts--python_2022-31-01-16-34-10_623843294/".
```
Comment (sulabhM) [inline test/suite/helper_tiered.py:89]: You can update the pattern example on line 89

**Review by sulabhM — APPROVED** [review]
I only glanced through changes to be aware of them. Looks good!

File: test/suite/helper_tiered.py (hunk 4 above)
Comment (ruby-oujo) [reply test/suite/helper_tiered.py:89]: Hi Sulabh, thanks for leaving your review. I have updated the pattern example.

**Pull request merged**